psst-gui/src/ui/public_user.rs

```diff
@@ -0,0 +1,368 @@
+            |_, _, _| {},
+            |_, data, (_username, r)| {
+                if let Err(err) = r {
+                    data.error_alert(err);
+                } else {
+                    data.info_alert("User unfollowed.");
+                    // Update the is_following state in the cached data
+                    if let Some(cached) = data.public_user_detail.info.resolved_mut() {
+                        let info = Arc::make_mut(&mut cached.data);
+                        info.is_following = Some(false);
+                        info.followers_count -= 1;
+                    }
+                }
+            },
+        )
+}
+
+fn loaded_detail_widget() -> impl Widget<WithCtx<Cached<Arc<PublicUserInformation>>>> {
+    let user_profile_top = user_info_widget().padding(theme::grid(1.0));
+
+    // This puts it as read only, it is needed to trainsform the context into what
+    // is needed
+    let user_playlists = user_playlists_widget().lens(Map::new(
+        |data: &WithCtx<Cached<Arc<PublicUserInformation>>>| WithCtx {
+            ctx: data.ctx.clone(),
+            data: data.data.data.public_playlists.clone(),
+        },
+        |_, _| {},
+    ));
+    let user_artists = user_artists_widget().lens(Map::new(
+        |data: &WithCtx<Cached<Arc<PublicUserInformation>>>| WithCtx {
+            ctx: data.ctx.clone(),
+            data: data.data.data.recently_played_artists.clone(),
+        },
+        |_, _| {},
+    ));
+    let user_followers = user_followers_widget().lens(Map::new(
+        |data: &WithCtx<Cached<Arc<PublicUserInformation>>>| WithCtx {
+            ctx: data.ctx.clone(),
+            data: data.data.data.followers.clone(),
+        },
+        |_, _| {},
+    ));
+    let user_following = user_following_widget().lens(Map::new(
+        |data: &WithCtx<Cached<Arc<PublicUserInformation>>>| WithCtx {
+            ctx: data.ctx.clone(),
+            data: data.data.data.following.clone(),
+        },
+        |_, _| {},
+    ));
+
+    Flex::column()
+        .cross_axis_alignment(CrossAxisAlignment::Start)
+        .with_spacer(theme::grid(1.0))
+        .with_child(user_profile_top)
+        .with_spacer(theme::grid(1.0))
+        .with_child(user_playlists)
+        .with_spacer(theme::grid(1.0))
+        .with_child(user_artists)
+        .with_spacer(theme::grid(1.0))
+        .with_child(user_followers)
+        .with_spacer(theme::grid(1.0))
+        .with_child(user_following)
+}
+
+pub fn cover_widget(size: f64) -> impl Widget<WithCtx<Cached<Arc<PublicUserInformation>>>> {
+    let radius = size / 2.0;
+
+    // Placeholder widget showing a user icon
+    let placeholder = icons::USER
+        .scale((size * 0.9, size * 0.9))
+        .with_color(Color::rgb8(0x53, 0x53, 0x53))
+        .padding((size * 0.05, size * 0.05))
+        .center()
+        .fix_size(size, size)
+        .background(Color::rgb8(0x28, 0x28, 0x28))
+        .clip(Circle::new((radius, radius), radius));
```

Comment:
Might be good to put this in our icons folder and then import? Maybe there is a nice SF Pro icon we can use?